hamocc/aufw_bgc.F90

```diff
@@ -164,13 +158,17 @@ SUBROUTINE AUFW_BGC(kpie,kpje,kpke,ntr,ntrbgc,itrbgc,trc,              &
       save  /xcmpii/
 #endif
 
-! pass tracer fields in from ocean model, note that both timelevels 
-! are passed into the local array locetra; No unit conversion here, 
-! tracers in the restart file are written in mol/kg 
+! pass tracer fields in from ocean model, note that both timelevels
+! are passed into the local array locetra; No unit conversion here,
+! tracers in the restart file are written in mol/kg
 !--------------------------------------------------------------------
 !
       testio=0
-      locetra(:,:,:,:)=trc(1:kpie,1:kpje,:,itrbgc:itrbgc+ntrbgc-1)
+!
+! Allocate and initialize local array for writing (locetra)
+!
+      locetra(:,:,:,:) = trc(1:kpie,1:kpje,:,itrbgc:itrbgc+ntrbgc-1)  ! TODO - is this right?
```

Comment:
This should be correct (the trc array carries 2 time-levels of tracer values in the 3rd dimension, these are both copied into locetra here). 

Not sure though - wasn't it necessary to have locetra as allocatable array? If it works it's fine

Reply:
@JorgSchwinger - locetra works now as a non-allocatable array on the stack.